sklearn/inspection/_ice.py

```diff
@@ -0,0 +1,491 @@
+    instances = X.shape[0]
+
+    # reshape to (n_targets, n_instances, n_points) where n_targets is:
+    # - 1 for non-multioutput regression and binary classification (shape is
+    #   already correct in those cases)
+    # - n_tasks for multi-output regression
+    # - n_classes for multiclass classification.
+    predictions = np.array(predictions).T
+    if is_regressor(estimator) and predictions.ndim == 2:
+        # non-multioutput regression, shape is (n_instances, n_points,)
+        predictions = predictions.reshape(instances, -1)
+    elif is_classifier(estimator) and predictions.shape[0] == 2:
+        # Binary classification, shape is (2, n_instances, n_points).
+        # we output the effect of **positive** class
+        predictions = predictions[1]
+        predictions = predictions.reshape(instances, -1)
+
+    return predictions
+
+
+def individual_conditional_expectation(estimator, X, features,
+                                       response_method='auto',
+                                       percentiles=(0.05, 0.95),
+                                       grid_resolution=100, centre=True,
+                                       **kwargs):
+    """Individual Conditional Expectation (ICE) of ``features``.
+
+    ICE of a feature (or a set of features) corresponds to the responses of an
+    estimator for each possible value of the feature for all instance in ``X``.
+
+    Read more in the :ref:`User Guide <individual_conditional_expectation>`.
+
+    Parameters
+    ----------
+    estimator : BaseEstimator
+        A fitted estimator object implementing :term:`predict`,
+        :term:`predict_proba`, or :term:`decision_function`.
+        Multioutput-multiclass classifiers are not supported.
+
+    X : {array-like or dataframe} of shape (n_samples, n_features)
+        ``X`` is used both to generate a grid of values for the
+        ``features``, and to compute the averaged predictions.
+
+    features : array-like of {int, str}
+        The feature (e.g. `[0]`) or pair of interacting features
+        (e.g. `[(0, 1)]`) for which the ICE should be computed.
+
+    response_method : 'auto', 'predict_proba' or 'decision_function', \
+            optional (default='auto')
+        Specifies whether to use :term:`predict_proba` or
+        :term:`decision_function` as the target response. For regressors
+        this parameter is ignored and the response is always the output of
+        :term:`predict`. By default, :term:`predict_proba` is tried first
+        and we revert to :term:`decision_function` if it doesn't exist.
+
+    percentiles : tuple of float, optional (default=(0.05, 0.95))
+        The lower and upper percentile used to create the extreme values
+        for the grid. Must be in [0, 1].
+
+    grid_resolution : int, optional (default=100)
+        The number of equally spaced points on the grid, for each target
+        feature.
+
+    centre : bool, optional (default=True)
```

Comment:
```suggestion
    center : bool, optional (default=True)
```

Reply:
Maybe this can have a more explicit name, like "start_at_origin" or "fixed_start_point"

Reply:
I went ahead with `fixed_start_point`. Done in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/fee74e12fb4d6519c4f0fa42c975553d845bd5bc